### PR TITLE
refactor: remove hardcoded tier 3 config

### PR DESCRIPTION
**Pull request by andrewHEguardian**
## What are you doing in this PR?

Remove remaining client side hardcoded tier 3 config. Follow on from #6261 

[**Trello Card**](https://trello.com/c/ZaGaRI9h/1013-unpick-and-remove-old-tier-3-checkout-changes)

### CODE REVIEW

File: support-frontend/assets/helpers/productCatalog.tsx [hunk 1]
```diff
@@ -312,244 +312,3 @@ export const productCatalogDescription: Record<ProductKey, ProductDescription> =
...
-		name: 'SupportPlusAndGuardianWeekly',
-		description: 'Supporter Plus and Guardian Weekly',
-		promoCode: '3TIER_WEEKLY_AU_MONTHLY_V2',
-		discountedPrice: 48,
-		discount: {
-			amount: 20.0,
-			durationMonths: 12,
-		},
-	},
-};
```
Comment (jamesgorrie) [inline support-frontend/assets/helpers/productCatalog.tsx:555]: 😌

File: support-frontend/assets/helpers/subscriptionsForms/submit.ts [hunk 2]
```diff
@@ -228,7 +219,7 @@ function buildRegularPaymentRequest(
 		csrUsername,
 		salesforceCaseId,
 		debugInfo: actionHistory,
-		threeTierCreateSupporterPlusSubscription: inThreeTier,
+		threeTierCreateSupporterPlusSubscription: false,
```
Comment (jamesgorrie) [inline support-frontend/assets/helpers/subscriptionsForms/submit.ts:222]: Should we not remove this completely?

**Review by jamesgorrie — APPROVED** [review]
Great, removed quite a lot of abstractions that were used for this!

File: support-frontend/assets/helpers/subscriptionsForms/submit.ts (hunk 2 above)
Comment (andrewHEguardian) [reply support-frontend/assets/helpers/subscriptionsForms/submit.ts:222]: Looks like it's actually optional [in the backend](https://github.com/guardian/support-frontend/blob/d678f7c635f346c97be132e986198ac20924c3fb/support-frontend/app/services/stepfunctions/SupportWorkersClient.scala#L66)? So we can safely remove it

**Pull request merged**